library/modules/EventManager.cpp

```diff
@@ -271,15 +296,15 @@ void DFHack::EventManager::onStateChange(color_ostream& out, state_change_event
             }
             constructions[constr->pos] = *constr;
         }
-        for ( size_t a = 0; a < df::global::world->buildings.all.size(); a++ ) {
+        for ( size_t a = 0; a < df::global::world->buildings.all.size(); ++a ) {
             df::building* b = df::global::world->buildings.all[a];
             Buildings::updateBuildings(out, (void*)intptr_t(b->id));
             buildings.insert(b->id);
         }
         lastSyndromeTime = -1;
-        for ( size_t a = 0; a < df::global::world->units.all.size(); a++ ) {
+        for ( size_t a = 0; a < df::global::world->units.all.size(); ++a ) {
```

Comment:
In cases where the loop variable is a primitive value (a `size_t` in this case), `++a` is _not_ any faster than `a++`.

However, in nearly all of these cases that I see in this PR, the use of iterators (or loop variables) could be replaced with a range-based for loop, such as:

```diff
-        for ( size_t a = 0; a < df::global::world->units.all.size(); ++a ) {
-            df::unit* unit = df::global::world->units.all[a];
+        for ( df::unit *unit : df::global::world->units.all ) {
```

This also works with iterators (maybe with slight changes for iterating over maps). If we're rewriting loops, I would prefer this approach.

Reply:
I'll update the loops soon. I've been doing a lot of data migration and permissions may have gotten bonked on more than one repo. Should be able to finish this by the end of August.

Reply:
Btw I just replaced all the post increments with find iirc. From my recollection pre is not always better but it is never worse, so I just use it exclusively and use those 3 brain cells for other tasks.